src/transform/src/union_cancel.rs

```diff
@@ -26,6 +26,22 @@ impl crate::Transform for UnionBranchCancellation {
     }
 }
 
+enum BranchCmp {
```

Comment:
Could you just add some comments as to what the three types of branch comparison mean for the sake of future reference?